src/server/system_services/account_server.js

```diff
@@ -691,73 +691,91 @@ async function add_external_connection(req) {
     });
 }
 
-async function edit_external_connection_credentials(req) {
-    const { name, identity, secret } = req.rpc_params;
-    const { account } = req;
-    const connection_to_edit = cloud_utils.find_cloud_connection(account, name);
-    const { endpoint_type, cp_code, auth_method, endpoint } = connection_to_edit;
-    const new_connection = _.cloneDeep(connection_to_edit);
-    const check_connection = {
+async function update_external_connection(req) {
+    const connection = cloud_utils.find_cloud_connection(req.account, req.rpc_params.name);
+    const {
         name,
-        identity,
-        secret,
-        endpoint_type,
-        cp_code,
-        auth_method,
-        endpoint,
-    };
+        identity = new SensitiveString(connection.access_key),
```

Comment:
What happens if you do send identity in the RPC, you're not checking it and not setting it later on ?

Reply:
It will take the value from the rpc_params and will not use the default value. 
It works as expected